### PR TITLE
Issue 73 my task page

### LINKED ISSUE #73: My Task Page
## Basic Information
Create a My Task page consisting of My Tasks, and My Bids. Both "view" should show a list of `<TaskCard>` components or an image if empty. Place in `/components/my-tasks.tsx`

image

image

## Other Information
uses component `<Button>, <Heading>, <BottomNav>, <TopNav>, <TaskCard>`

### PR DESCRIPTION
**Pull request by ErikaKK**
## Change Summary
image
image

Using client-side data fetching to get task data from API. 

### Change Form
**Fill this up (NA if not available). If a certain criteria is not met, can you please give a reason.**

- [ ] The pull request title has an issue number
- [ ] The change works by "Smoke testing" or quick testing
- [ ] The change has tests
- [ ] The change has documentation

## Other Information
**[Is there anything in particular in the review that I should be aware of?]**
The data used to filter task will need to change later. Currently I use username to do the test.

# Related issue

- Resolve #73

### CODE REVIEW

**Comment by ErikaKK** [comment]
I'll add sorting to the page

**Comment by yunho7687** [comment]
@HuxleyBerry  could you help review this PR please 🙏

**Review by yunho7687 — APPROVED** [review]
lgtm

File: client/src/components/ui/bottom-nav.tsx [hunk 1]
```diff
@@ -53,7 +53,7 @@ const BottomNav = ({
 
   return (
     <div
-      className={`border-t-penni-border-light-mode ${isFixed ? "fixed bottom-0" : ""} h-20 w-full border-t-2`}
+      className={`border-t-penni-border-light-mode ${isFixed ? "fixed bottom-0" : ""} h-20 w-full border-t-2 bg-white`}
```
Comment (SodaVolcano) [inline client/src/components/ui/bottom-nav.tsx]: Replace `bg-white` with `bg-penni-background-light-mode`

File: client/src/components/ui/top-navtab.tsx [hunk 2]
```diff
@@ -7,14 +7,15 @@ interface Tab {
...
 }
 
-export default function TopNavtab({ tabs }: TopNavtabProps) {
+export default function TopNavtab({ tabs, isFixed = false }: TopNavtabProps) {
   const [activeTab, setActiveTab] = useState(tabs[0].name);
 
   const handleTabClick = (tabName: string) => {
     setActiveTab(tabName);
-    console.log(`Active tab is now: ${tabName}`);
+    // console.log(`Active tab is now: ${tabName}`);
```
Comment (SodaVolcano) [inline client/src/components/ui/top-navtab.tsx]: Remove, don't need to comment out code - we can look through past commits if we ever need the code snippet in the future.

File: client/src/components/ui/top-navtab.tsx [hunk 3]
```diff
@@ -23,10 +24,20 @@ export default function TopNavtab({ tabs }: TopNavtabProps) {
...
+      setPaddingTop(targetDivRef.current.offsetHeight);
+    }
+  }, []);
   return (
-    <div className="flex w-full flex-col">
-      <div className="flex border-b">
+    <div className="relative flex w-full flex-col">
+      <div
+        ref={targetDivRef}
+        className={`${isFixed ? "fixed top-0" : ""} z-40 flex w-full border-b bg-white`}
```
Comment (SodaVolcano) [inline client/src/components/ui/top-navtab.tsx]: Should `bg-white` be `bg-penni-background-light-mode`?

**Review by SodaVolcano — CHANGES_REQUESTED** [review]
Very cool and epic and rad and neat and lit and

File: client/src/components/ui/top-navtab.tsx (hunk 3 above)
Comment (yunho7687) [reply client/src/components/ui/top-navtab.tsx]: the latter is better, I reckon

File: client/src/components/ui/bidder-mytasks-page.tsx [hunk 4]
```diff
@@ -0,0 +1,138 @@
+import React, { useEffect, useState } from "react";
```
Comment (HuxleyBerry) [inline client/src/components/ui/bidder-mytasks-page.tsx]: You don't need to import React, I believe nextjs does it automatically

File: client/src/components/ui/top-navtab.tsx [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import React, { useState } from "react";
+import React, { useEffect, useRef, useState } from "react";
```
Comment (HuxleyBerry) [inline client/src/components/ui/top-navtab.tsx]: Same

**Review by HuxleyBerry — CHANGES_REQUESTED** [review]
Looks good, just a couple of minor things I noticed

File: client/src/components/ui/bidder-mytasks-page.tsx [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+}
+
+// using user name and state to filter task
+const filterTasks = (tasks: Task[], name: string, states: string[]) => {
+  return tasks.filter(
+    (task) => task.name === name && states.includes(task.state),
+  );
+};
+
+const TaskList = ({ name, state }: TaskListProps) => {
```
Comment (HuxleyBerry) [inline client/src/components/ui/bidder-mytasks-page.tsx]: Per the style guide, use ``function`` syntax for components, rather than an arrow function.

**Review by HuxleyBerry — APPROVED** [review]

**Review by SodaVolcano — APPROVED** [review]

**Pull request merged**